src/redux/CarsList/reducer.ts

```diff
@@ -0,0 +1,17 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { CarDetails } from "../../models/CarDetails";
+
+const initialState: { cars: CarDetails[] } = { cars: [] };
```

Comment:
Now its crucial to represent the states correctly, I would propose the following state convention for everyone
by default/ initial state can be 
```
{ cars: null, loading: null} // no cars as of now, we haven't even hit the api
{cars: null, loading: true} // no cars yet, but we are loading the data
{cars:[],loading:false} // no results, or no cars present or found
{cars:[1,2,3], loading:false} // 3 cars found after loading
```
Its important to have/consider the loading state
Something similar to what has been given here: https://stackoverflow.com/a/56466086
☝️ this one is a bit advanced and uses constants to maintain those states, use this as a reference.